### PR TITLE
=str Reduce allocation of throttle.

### PR DESCRIPTION
**Pull request by He-Pin**
A little clean up.

### CODE REVIEW

File: akka-stream/src/main/scala/akka/stream/impl/Throttle.scala [hunk 1]
```diff
@@ -45,57 +46,43 @@ import akka.util.NanoTimeTokenBucket
...
+    new TimerGraphStageLogic(shape) with InHandler with OutHandler {
+      private val tokenBucket = new NanoTimeTokenBucket(effectiveMaximumBurst, nanosBetweenTokens)
+      private var currentElement: T = _
 
-    var willStop = false
-    var currentElement: T = _
-    val enforcing = mode match {
-      case Enforcing => true
-      case Shaping   => false
-    }
```
Comment (hepin1989) [inline akka-stream/src/main/scala/akka/stream/impl/Throttle.scala:58]: reduce some allocation here

**Review by johanandren — APPROVED** [review]
LGTM, nice little simplification/optimization

**Comment by hepin1989** [comment]
I have a lot of concurrent streams:)

**Comment by johanandren** [comment]
I saw the tweet :) 👍

File: akka-stream/src/main/scala/akka/stream/impl/Throttle.scala [hunk 2]
```diff
@@ -4,20 +4,21 @@
...
 import akka.util.NanoTimeTokenBucket
 
+import scala.concurrent.duration.{ FiniteDuration, _ }
+
 /**
  * INTERNAL API
  */
 @InternalApi private[akka] object Throttle {
   final val AutomaticMaximumBurst = -1
+  private val timerName: String = "ThrottleTimer"
```
Comment (patriknw) [inline akka-stream/src/main/scala/akka/stream/impl/Throttle.scala:21]: that can be
```
private case object TimerKey
```

**Review by patriknw — APPROVED** [review]
LGTM, with a small suggestion

File: akka-stream/src/main/scala/akka/stream/impl/Throttle.scala (hunk 2 above)
Comment (hepin1989) [reply akka-stream/src/main/scala/akka/stream/impl/Throttle.scala:21]: Will update tomorrow

**Comment by hepin1989** [comment]
@patriknw I have updated it with `case object TimerKey`

**Pull request merged**